src/game_io/game_preload_packet.cc

```diff
@@ -52,11 +52,13 @@ void GamePreloadPacket::read(FileSystem& fs, Game&, MapObjectLoader* const) {
 		Section& s = prof.get_safe_section("global");
 		int32_t const packet_version = s.get_int("packet_version");
 
-		if (packet_version == kCurrentPacketVersion) {
+		if (packet_version >= 6 && packet_version <= kCurrentPacketVersion) {
 			gametime_ = s.get_safe_int("gametime");
 			mapname_ = s.get_safe_string("mapname");
 
 			background_ = s.get_safe_string("background");
+			// TODO(Nordfriese): Savegame compatibility
+			background_theme_ = (packet_version < 7 ? "" : s.get_safe_string("theme"));
```

Comment:
If the packet has a "world" string, map that to the theme.

```
blackland = blackland
greenland = summer
winterland = winter
desert = desert
```

Also look for consistency with the random map UI. where we have:
```

     world_descriptions_({
        /** TRANSLATORS: A world name for the random map generator in the editor */
        {"greenland", _("Summer")},
        /** TRANSLATORS: A world name for the random map generator in the editor */
        {"winterland", _("Winter")},
        /** TRANSLATORS: A world name for the random map generator in the editor */
        {"desert", _("Desert")},
        /** TRANSLATORS: A world name for the random map generator in the editor */
        {"blackland", _("Wasteland")},
     }),
```